### PR TITLE
DOCS-223 CLC Overview

### PR DESCRIPTION
**Pull request by angelasimms**
Overview article - Early draft for feedback

### CODE REVIEW

File: docs/modules/ROOT/pages/overview.adoc [hunk 1]
```diff
@@ -1,14 +1,97 @@
...
+
+== Modes
+
+Hazelcast CLC works in three modes:
+- Non-Interactive
+- Interactive
+- SQL
+
+You can easily switch between modes to suit the context in which you're working.
+// Is this true, can you easily disable interactive mode? I couldn't.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:37]: I assume you just use `exit` to get out of interactive mode. @utku-caglayan

File: docs/modules/ROOT/pages/overview.adoc [hunk 2]
```diff
@@ -1,14 +1,97 @@
...
+[source,bash,subs="attributes+"]
+----
+hzc sql
+----
 
 == Limitations
 
-- *Client Authentication:* Hazelcast CLC supports only simple authentication for Hazelcast Platform. Hazelcast CLC does not currently support any other client authentication.
\ No newline at end of file
+*Client authentication:* Hazelcast CLC supports only simple authentication for Hazelcast Platform. Hazelcast CLC does not currently support any other client authentication.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:90]:
```suggestion
*Client authentication:* Hazelcast CLC supports only simple authentication (username and password) for Hazelcast Platform. Hazelcast CLC does not currently support any other client authentication. If your Hazelcast Platform cluster uses other methods of client authentication, you cannot use Hazelcast CLC to connect to it.
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 3]
```diff
@@ -1,14 +1,97 @@
...
+
+== Use Cases
+
+Example use cases for Hazelcast CLC.
+
+=== Basic Cluster Management for Hazelcast Platform
+
+Perform xref:hzc-cluster.adoc[administration tasks] on multiple clusters without writing code. Instead, use <<interactive-mode, interactive mode>> to explore available commands.
+
+Check or change the state of a cluster, update a configuration option, then switch to another cluster direct from the command line. For maximum flexibility, you can set cluster connection details at the command line or using a xref:configuration.adoc[custom configuration file].
```
Comment (utku-caglayan) [inline docs/modules/ROOT/pages/overview.adoc:17]:
```suggestion
Check or change the state of a cluster, update a configuration option, then switch to another cluster directly from the command line. For maximum flexibility, you can set cluster connection details at the command line or using a xref:configuration.adoc[custom configuration file]. 
```

File: docs/modules/ROOT/pages/overview.adoc (hunk 1 above)
Comment (utku-caglayan) [reply docs/modules/ROOT/pages/overview.adoc:37]: Yes, from interactive mode one can exit by typing "exit" or pressing Ctrl+c together. Also you can switch to SQL Browser by typing "sql" from the interactive mode.

File: docs/modules/ROOT/pages/overview.adoc [hunk 4]
```diff
@@ -1,14 +1,97 @@
 = Hazelcast Command-Line Client (CLC)
-:description: Short description with main points
+:url-github-clc: https://github.com/hazelcast/hazelcast-cloud-cli/blob/master/README.md 
+:description: You can use Hazelcast Command Line Client (CLC) to connect to and interact with clusters on Hazelcast Platform and Hazelcast Viridian, direct from the command line or through scripts.
```
[the remark this answers is not in the document]
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:3]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 1 above)
Comment (JakeSCahill) [reply docs/modules/ROOT/pages/overview.adoc:37]: I think this is related to an issue #96

File: docs/modules/ROOT/pages/overview.adoc (hunk 4 above)
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:3]:
```suggestion
:description: You can use the Hazelcast Command Line Client (CLC) to connect to and interact with clusters on Hazelcast Platform and Hazelcast Viridian, direct from the command line or through scripts.
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 5]
```diff
@@ -1,14 +1,97 @@
...
-What does the product do?
-what are some examples of where you’d use it?
-Who is it for?
-What problem does the product solve?
-What modes are available and what are they?
-What (if any) limitations does the product have?
-How do you get started?
+{description}
+
+Hazelcast CLC simplifies the client setup process for Unix-like local environments. Within minutes of installation, you can start to perform common tasks on clusters.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:7]:
```suggestion
The Hazelcast CLC simplifies the client setup process. Within minutes of installation, you can start to perform common tasks on clusters.
```

It looks like we are adding an install script for Windows.

File: docs/modules/ROOT/pages/overview.adoc [hunk 6]
```diff
@@ -1,14 +1,97 @@
...
+
+== Use Cases
+
+Example use cases for Hazelcast CLC.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:11]:
```suggestion
Example use cases for the Hazelcast CLC.
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 7]
```diff
@@ -1,14 +1,97 @@
...
+
+Get xref:hzc-map.adoc[direct access to map data] for quick debugging of prototypes or production applications. Easily read or query data on a cluster using <<sql-mode, SQL mode>> to view query results, and then perform updates without changing tools.
+
+=== Scripting for Automation
+
+Work in <<non-interactive-mode, non-interactive mode>> to script and execute repetitive administration, integration, or testing tasks.
+
+== Modes
+
+Hazelcast CLC works in three modes:
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:31]:
```suggestion
The Hazelcast CLC works in three modes:
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 8]
```diff
@@ -1,14 +1,97 @@
...
+- Interactive
+- SQL
+
+You can easily switch between modes to suit the context in which you're working.
+// Is this true, can you easily disable interactive mode? I couldn't. 
+
+[[non-interactive-mode]]
+=== Non-Interactive Mode
+
+This is the default mode for Hazelcast CLC. You need to manually enter and execute commands at the command line. This mode is useful for scripting.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:42]:
```suggestion
This is the default mode for the Hazelcast CLC. You need to manually enter and execute commands at the command line. This mode is useful for scripting.
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 9]
```diff
@@ -1,14 +1,97 @@
...
+hzc
+----
+
+[[sql-mode]]
+=== SQL Mode
+You can execute SQL statements in all modes, but SQL mode also gives you access to an interactive SQL browser to navigate query results.
+
+image:ROOT:hzc-sql-browser-screenshot.png[HZC interactive]
+
+To start Hazelcast CLC in SQL mode, do the following:
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:81]:
```suggestion
To start the Hazelcast CLC in SQL mode, do the following:
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 10]
```diff
@@ -1,14 +1,97 @@
...
+----
 
 == Limitations
 
-- *Client Authentication:* Hazelcast CLC supports only simple authentication for Hazelcast Platform. Hazelcast CLC does not currently support any other client authentication.
\ No newline at end of file
+*Client authentication:* Hazelcast CLC supports only simple authentication for Hazelcast Platform. Hazelcast CLC does not currently support any other client authentication.
+
+== Next Steps
+Install, configure, and explore Hazelcast CLC:
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/overview.adoc:93]:
```suggestion
Install, configure, and explore the Hazelcast CLC:
```

**Review by JakeSCahill — CHANGES_REQUESTED** [review]
The CLC will support Windows through an install script. I also added articles before 'Hazelcast CLC' - since it's an industry-wide term, we shouldn't customize it. Lots of other companies use 'the ... CLI'.

File: docs/modules/ROOT/pages/overview.adoc (hunk 1 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:37]: I have added the ways to exit this mode and will track #96.

File: docs/modules/ROOT/pages/overview.adoc (hunk 4 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:3]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 5 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:7]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 6 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:11]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 7 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:31]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 8 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:42]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 9 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:81]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 10 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:93]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 2 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:90]: Updated as requested.

File: docs/modules/ROOT/pages/overview.adoc (hunk 3 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:17]: Reworded slightly for easier reading.

File: docs/modules/ROOT/pages/overview.adoc (hunk 5 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/overview.adoc:7]: Agreed to leave pending merging of PR for Windows version.

**Review by JakeSCahill — APPROVED** [review]

File: docs/modules/ROOT/pages/overview.adoc (hunk 1 above)
Comment (utku-caglayan) [reply docs/modules/ROOT/pages/overview.adoc]: Oh I see, sorry to hear :(. I will try to reproduce the mentioned issue. Thanks

File: docs/modules/ROOT/pages/overview.adoc [hunk 11]
```diff
@@ -1,14 +1,104 @@
...
+----
+
+You can xref:install-clc.adoc[enable interactive mode] for bash and zsh shells after installation. To start Hazelcast CLC in interactive mode, do the following:
+
+[source,bash,subs="attributes+"]
+----
+hzc
+----
+
+To return to non-interactive mode, use either of the following methods:
```
Comment (utku-caglayan) [inline docs/modules/ROOT/pages/overview.adoc:75]:
```suggestion
To exit, use either of the following methods:
```
I think this one is correct

File: docs/modules/ROOT/pages/overview.adoc [hunk 12]
```diff
@@ -1,14 +1,104 @@
...
+
+Example use cases for the Hazelcast CLC.
+
+=== Basic Cluster Management for Hazelcast Platform
+
+Perform xref:hzc-cluster.adoc[administration tasks] on multiple clusters without writing code. Instead, use <<interactive-mode, interactive mode>> to explore available commands.
+
+Check or change the state of a cluster, update a configuration option, then switch to another cluster, all from the command line. For maximum flexibility, you can set cluster connection details at the command line or using a xref:configuration.adoc[custom configuration file]. 
+
+NOTE: {hazelcast-cloud} has a separate link:{url-github-clc}[CLI for cluster management].
```
Comment (utku-caglayan) [inline docs/modules/ROOT/pages/overview.adoc:19]: I can not see the image on preview deployment. Maybe it is expected but just wanted to point out
https://deploy-preview-95--eclectic-sawine-19fcf1.netlify.app/docs/clc/0.1/overview.html

**Review by utku-caglayan — APPROVED** [review]
Looks good thanks!

**Pull request merged**